### PR TITLE
Add sudo endpoint to APIs and APPs

### PR DESCRIPTION
**Pull request by CyberHoward**
Add sudo endpoint to the modules
also fixes the missing reply handlers for the API contract. 

Cleans up some of the traits in abstract_sdk base as well.

### CODE REVIEW

**Comment by adairrr** [comment]
Task linked: [CU-861mkcb4f Add sudo endpoint to modules](https://app.clickup.com/t/861mkcb4f)

**Comment by adairrr** [comment]
Task linked: [CU-861mkcb52 Add reply endpoint to API](https://app.clickup.com/t/861mkcb52)

File: packages/abstract-api/src/endpoints/ibc_callback.rs [hunk 1]
```diff
@@ -1,13 +1,64 @@
...
+// to test this, add ibc client to version control
+
+// #[cfg(test)]
+// mod tests {
+//     use abstract_core::{api::{ExecuteMsg, BaseExecuteMsg, self}, abstract_ica::IbcResponseMsg};
+//     use abstract_sdk::base::ExecuteEndpoint;
+//     use abstract_testing::prelude::{TEST_MANAGER, mocked_account_querier_builder};
+//     use cosmwasm_std::{testing::{mock_dependencies, mock_env, mock_info}, DepsMut, Response};
+//     use speculoos::prelude::*;
+//     use crate::mock::{ApiMockResult, MockReceiveMsg, execute, MOCK_API, mock_init, MockExecMsg, MockError};
```
Comment (adairrr) [inline packages/abstract-api/src/endpoints/ibc_callback.rs:19]: Are we saving this for later or can we remove it?

File: packages/abstract-api/src/handler.rs [hunk 2]
```diff
@@ -1,33 +1,19 @@
...
-
     type CustomMigrateMsg = Empty;
-
+    type SudoMsg = SudoMsg;
     type ReceiveMsg = Receive;
 
-    fn contract(
-        &self,
-    ) -> &AbstractContract<Self, Error, InitMsg, ExecMsg, QueryMsg, Empty, Receive> {
+    fn contract(&self) -> &AbstractContract<Self, Error> {
```
Comment (adairrr) [inline packages/abstract-api/src/handler.rs:16]: Are the entry-point types inferred here?

File: packages/abstract-app/src/endpoints.rs [hunk 3]
```diff
@@ -67,6 +68,17 @@ macro_rules! export_endpoints {
...
+        // Sudo entrypoint
+        #[::cosmwasm_std::entry_point]
+        pub fn sudo(
+            deps: ::cosmwasm_std::DepsMut,
+            env: ::cosmwasm_std::Env,
+            msg: <$app_type as ::abstract_sdk::base::Handler>::SudoMsg,
+        ) -> Result<::cosmwasm_std::Response, <$app_type as ::abstract_sdk::base::Handler>::Error> {
+            use ::abstract_sdk::base::SudoEndpoint;
+            $app_const.sudo(deps, env, msg)
+        }
```
Comment (adairrr) [inline packages/abstract-app/src/endpoints.rs:81]: We should add a basic test to this below

**Review by adairrr — COMMENTED** [review]
Just some small comments. Otherwise lgtm!

File: packages/abstract-api/src/endpoints/ibc_callback.rs (hunk 1 above)
Comment (CyberHoward) [reply packages/abstract-api/src/endpoints/ibc_callback.rs:19]: we should save it imo for after IBC revamp

File: packages/abstract-api/src/handler.rs (hunk 2 above)
Comment (CyberHoward) [reply packages/abstract-api/src/handler.rs:16]: They are inferred through the `Handler` trait!

Comment (CyberHoward) [reply packages/abstract-api/src/handler.rs:16]: see the function handler type definitions in the AbstractContract type

File: packages/abstract-app/src/endpoints.rs (hunk 3 above)
Comment (CyberHoward) [reply packages/abstract-app/src/endpoints.rs:81]: ahaa yes, those are the tests that you added in Barcelona, completely forgot about them

File: packages/abstract-api/src/handler.rs (hunk 2 above)
Comment (adairrr) [reply packages/abstract-api/src/handler.rs:16]: Amazing. I love rust.

File: packages/abstract-api/src/endpoints/ibc_callback.rs (hunk 1 above)
Comment (adairrr) [reply packages/abstract-api/src/endpoints/ibc_callback.rs:19]: We should comment on any commented-out code that with a TODO or FIXME to make sure that we are aware of its purpose.

**Review by adairrr — APPROVED** [review]

**Comment by adairrr** [comment]
LGTM. Nice!

**Pull request merged**